lib/at_host/at_host.c

```diff
@@ -168,34 +196,54 @@ int at_host_init(enum select_uart uart_id, enum term_modes mode)
 		}
 	}
 
+	k_sem_give(&at_host_start_sem);
```

Comment:
Hmm, instead of passing the semaphore around you could just spawn the thread at this point. See how it's done in 15.4 driver:
https://github.com/zephyrproject-rtos/zephyr/blob/f0120848244be7d88f0b831c76e9e2da43f7dbde/drivers/ieee802154/ieee802154_nrf5.c#L368